src/QMCWaveFunctions/Fermion/SlaterDetWithBackflow.h

```diff
@@ -31,17 +31,20 @@ class SlaterDetWithBackflow : public SlaterDet
    * @param targetPtcl target Particleset
    * @param rn release node
    */
-  SlaterDetWithBackflow(ParticleSet& targetPtcl, std::shared_ptr<BackflowTransformation> BF);
+  SlaterDetWithBackflow(ParticleSet& targetPtcl, std::vector<std::unique_ptr<Determinant_t>> dets, std::shared_ptr<BackflowTransformation> BF);
 
   ///destructor
   ~SlaterDetWithBackflow() override;
 
   ///set BF pointers
-  void setBF(std::shared_ptr<BackflowTransformation> bf) override
+  void setBF(std::shared_ptr<BackflowTransformation> bf)
   {
     BFTrans = bf;
     for (int i = 0; i < Dets.size(); i++)
       Dets[i]->setBF(BFTrans);
+
+    if (BFTrans->isOptimizable())
```

Comment:
This seems different from the semantics of Optimizable in the SlaterDet base class. What if none of the dets are optimizable, this still makes sense?